### DISCUSSION TITLE
Use supabase backend with custom endpoints

### DISCUSSION THREAD

**Discussion by robsannaa** [supabase/supabase#5206] [Questions] [answered]
Hi all,

I am asking this question since I did not find any information in the docs, in addition, I would consider myself a newbie in the backend world.

I have a Node.js server with an endpoint that calls an external API, gets some data, processes them, and saves it to the database/sends it to the client.
This endpoint has some business logic implemented in it, hence it has its codebase to achieve the goal.
My question is: would it be possible, to use supabase to take care of the DB, auth, and so on to define custom(s) endpoints with their own logic, all of this by using the authentication to protect my custom endpoints? 

Thanks a lot

**Comment by GaryAustin1** [comment, marked as answer]
Sort of unclear what you are asking for exactly, but you can use postgres functions to carry out simple business logic.  They can be called from the api with an rpc call and honor RLS as you desire.  You can also have them trigger on changes to tables to carry out other logic.
You can even call external api's based on triggers or rpc functions using Supabase Function Hooks or the http extension.

**Reply by robsannaa** [reply]
Hi Gary, thanks a lot for your answer and sorry for not being clear.
I think you answered my question but to make it clear let me maybe walk you through an example:

I have a frontend in Next.js, the user fills a form with some input data the data then is posted to my current backend (Node.js) to an endpoint such as `/api/search`.

Once the data reaches the endpoint, the data of the request is used for 2 different tasks:
- Save it to the DB (currently MongoDB)
- Call an external API with the data in the body of the request, and receive (from the external API) some additional information. Which then is being appended to the newly created record (something like `findOneAndUpdate()` for MongoDB).

My question is, since the logic to parse, process, clean the response from the external API has its own "block of code" is it possible, using supabase, to declare a custom endpoint containing arbitrary custom logic?

Again, thanks for the patience!

**Reply by GaryAustin1** [reply]
Sounds like my answer applies.  Lookup Postgres function on the web for an ideas of what they can do.

**Reply by robsannaa** [reply]
Thanks again, Gary. Best!
